### PR TITLE
New Resource: `azurerm_advisor_suppression`

### PR DESCRIPTION
**Pull request by jkroepke**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

This resources takes care of advisor suppressions.

API Ref: https://learn.microsoft.com/en-us/rest/api/advisor/suppressions?view=rest-advisor-2023-01-01

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/advisor '-run=TestAccAnalysisServicesServer_basic' -timeout 300m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAnalysisServicesServer_basic
=== PAUSE TestAccAnalysisServicesServer_basic
=== CONT  TestAccAnalysisServicesServer_basic
--- PASS: TestAccAnalysisServicesServer_basic (39.60s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/advisor	39.703s
```

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* New Resource: `azurerm_advisor_suppression` [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @jkroepke - Thanks for this PR. We are moving away from this style of resource for new resources in favour of "Typed" resource as per the contributor docs here: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#typed-vs-untyped-resources
can you take a look at implementing this as a Typed resource?

Additionally, we'd need this PR splitting into separate PRs for the multiple purposes here, i.e. one for updating the API version, one for the additional property support, and of course this one for the new resource.

If you can take a look at those changes, we can continue review.

Thanks

**Comment by jkroepke** [comment]
#26205

**Comment by jackofallops** [comment]
> Hey @jackofallops do you have any example for where a service directory contains both styles of resources?

Hi @jkroepke - Quite a few services make use of both types at present, for example: [`applicationinsights`](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/applicationinsights/registration.go#L13-L16) and [`containers`](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/registration.go#L15-L18)
Hope that helps?

**Comment by jkroepke** [comment]
additional property support: #26220

**Comment by jkroepke** [comment]
Hey @jackofallops

I may need support here. I rewrite the resource based on the given example, but terraform says on the acctests, that the resource does not exists

File: website/docs/r/advisor_suppresion.html.markdown [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+* `name` - (Required) The Name which should be used for this Advisor suppression. Changing this forces a new Advisor suppression to be created.
+
+* `recommendation_id` - (Required) The ID of the Advisor recommendation to suppress. Changing this forces a new Advisor suppression to be created.
+
+* `resource_id` - (Required) The ID of the Resource to suppress the Advisor recommendation for. Changing this forces a new Advisor suppression to be created.
+
+---
+
+* `ttl` - (Optional) A optional time to live value. If omitted, the suppression will not expire. Changing this forces a new Advisor suppression to be created.
```
Comment (katbyte) [inline website/docs/r/advisor_suppresion.html.markdown:44]: what units is this in? seconds? if so should this be
```suggestion
* `ttl_in_seconds` - (Optional) A optional time to live value. If omitted, the suppression will not expire. Changing this forces a new Advisor suppression to be created.
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/advisor_suppresion.html.markdown (hunk 1 above)
Comment (jkroepke) [reply website/docs/r/advisor_suppresion.html.markdown:44]: The API of Azure expects this format: `HH:MM:SS` or `DD:HH:MM:SS`.

Should I offer a go duration format for end users and the resource take care of the format that the API expects? Or is it fine the document, that this is the expected format?

**Comment by jkroepke** [comment]
@katbyte i'm for an response

File: website/docs/r/advisor_suppresion.html.markdown (hunk 1 above)
Comment (katbyte) [reply website/docs/r/advisor_suppresion.html.markdown:44]:
```suggestion
* `ttl` - (Optional) A optional time to live value in the format `HH:MM:SS` or `DD:HH:MM:SS`. If omitted, the suppression will not expire. Changing this forces a new Advisor suppression to be created.
```

**Review by katbyte — APPROVED** [review]
Sorry for the delay @jkroepke - i think describing the format is fine, i've done so and LGTM now 🔬

**Pull request merged**